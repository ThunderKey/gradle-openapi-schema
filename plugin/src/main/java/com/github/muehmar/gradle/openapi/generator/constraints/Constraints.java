package com.github.muehmar.gradle.openapi.generator.constraints;

import java.util.Objects;
import java.util.Optional;
import java.util.function.Consumer;

public class Constraints {
  private final Min min;
  private final Max max;
  private final DecimalMin decimalMin;
  private final DecimalMax decimalMax;
  private final Size size;
  private final Pattern pattern;
  private final Email email;

  private Constraints(
      Min min,
      Max max,
      DecimalMin decimalMin,
      DecimalMax decimalMax,
      Size size,
      Pattern pattern,
      Email email) {
    this.min = min;
    this.max = max;
    this.decimalMin = decimalMin;
    this.decimalMax = decimalMax;
    this.size = size;
    this.pattern = pattern;
    this.email = email;
  }

  public static Constraints empty() {
    return new Constraints(null, null, null, null, null, null, null);
  }

  public static Constraints ofMin(Min min) {
    return Constraints.empty().withMin(min);
  }

  public static Constraints ofMax(Max max) {
    return Constraints.empty().withMax(max);
  }

  public static Constraints ofMinAndMax(Min min, Max max) {
    return Constraints.empty().withMin(min).withMax(max);
  }

  public static Constraints ofSize(Size size) {
    return Constraints.empty().withSize(size);
  }

  public static Constraints ofPattern(Pattern pattern) {
    return Constraints.empty().withPattern(pattern);
  }

  public static Constraints ofEmail() {
    return Constraints.empty().withEmail(new Email());
  }

  public static Constraints ofDecimalMin(DecimalMin decimalMin) {
    return Constraints.empty().withDecimalMin(decimalMin);
  }

  public static Constraints ofDecimalMax(DecimalMax decimalMax) {
    return Constraints.empty().withDecimalMax(decimalMax);
  }

  public Constraints withMin(Min min) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withMax(Max max) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withSize(Size size) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withPattern(Pattern pattern) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withEmail(Email email) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withDecimalMin(DecimalMin decimalMin) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withDecimalMax(DecimalMax decimalMax) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public void onMin(Consumer<Min> onMin) {
    Optional.ofNullable(min).ifPresent(onMin);
  }

  public void onMax(Consumer<Max> onMax) {
    Optional.ofNullable(max).ifPresent(onMax);
  }

  public void onDecimalMin(Consumer<DecimalMin> onDecimalMin) {
    Optional.ofNullable(decimalMin).ifPresent(onDecimalMin);
  }

  public void onDecimalMax(Consumer<DecimalMax> onDecimalMax) {
    Optional.ofNullable(decimalMax).ifPresent(onDecimalMax);
  }

  public void onSize(Consumer<Size> onSize) {
    Optional.ofNullable(size).ifPresent(onSize);
  }

  public void onPattern(Consumer<Pattern> onPattern) {
    Optional.ofNullable(pattern).ifPresent(onPattern);
  }

  public void onEmail(Consumer<Email> onEmail) {
    Optional.ofNullable(email).ifPresent(onEmail);
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Constraints that = (Constraints) o;
    return Objects.equals(min, that.min)
        && Objects.equals(max, that.max)
        && Objects.equals(size, that.size)
        && Objects.equals(pattern, that.pattern)
        && Objects.equals(email, that.email);
  }

  @Override
  public int hashCode() {
    return Objects.hash(min, max, size, pattern, email);
  }

  @Override
  public String toString() {
    return "Constraints{"
        + "min="
        + min
        + ", max="
        + max
        + ", size="
        + size
        + ", pattern="
        + pattern
        + ", email="
        + email
        + '}';
  }
}
