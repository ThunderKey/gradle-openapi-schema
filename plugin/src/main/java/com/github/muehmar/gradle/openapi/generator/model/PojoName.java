package com.github.muehmar.gradle.openapi.generator.model;

import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class PojoName {
  private final Name name;
  private final String suffix;

  private PojoName(Name name, String suffix) {
    this.name = name;
    this.suffix = suffix;
  }

  public static PojoName ofName(Name name) {
    return new PojoName(name, "");
  }

  public static PojoName ofNameAndSuffix(Name name, String suffix) {
    return new PojoName(name, suffix);
  }

  public static PojoName ofNameAndSuffix(String name, String suffix) {
    return new PojoName(Name.ofString(name), suffix);
  }

  public static PojoName deriveOpenApiPojoName(PojoName pojoName, Name pojoMemberName) {
    final Name name = pojoName.getName().startUpperCase().append(pojoMemberName.startUpperCase());
    return new PojoName(name, pojoName.getSuffix());
  }

  public boolean equalsIgnoreCase(PojoName other) {
    return name.equalsIgnoreCase(other.name);
  }

  public PojoName startUppercase() {
    return new PojoName(name.startUpperCase(), suffix);
  }

  public Name getName() {
    return name;
  }

  public String getSuffix() {
    return suffix;
  }

  public String asString() {
    return name.append(suffix).asString();
  }
}
