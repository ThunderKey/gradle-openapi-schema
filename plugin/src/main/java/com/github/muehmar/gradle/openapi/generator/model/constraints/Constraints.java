package com.github.muehmar.gradle.openapi.generator.model.constraints;

import com.github.muehmar.gradle.openapi.util.Optionals;
import io.github.muehmar.pojobuilder.annotations.PojoBuilder;
import java.util.Optional;
import java.util.function.Function;
import lombok.Value;

@Value
@PojoBuilder
public class Constraints {
  Optional<Min> min;
  Optional<Max> max;
  Optional<DecimalMin> decimalMin;
  Optional<DecimalMax> decimalMax;
  Optional<Size> size;
  Optional<Pattern> pattern;
  Optional<Email> email;
  Optional<PropertyCount> propertyCount;
  Optional<MultipleOf> multipleOf;

  public static Constraints empty() {
    return new Constraints(
        Optional.empty(),
        Optional.empty(),
        Optional.empty(),
        Optional.empty(),
        Optional.empty(),
        Optional.empty(),
        Optional.empty(),
        Optional.empty(),
        Optional.empty());
  }

  public static Constraints ofMin(Min min) {
    return Constraints.empty().withMin(min);
  }

  public static Constraints ofMax(Max max) {
    return Constraints.empty().withMax(max);
  }

  public static Constraints ofMinAndMax(Min min, Max max) {
    return Constraints.empty().withMin(min).withMax(max);
  }

  public static Constraints ofSize(Size size) {
    return Constraints.empty().withSize(size);
  }

  public static Constraints ofPattern(Pattern pattern) {
    return Constraints.empty().withPattern(pattern);
  }

  public static Constraints ofEmail() {
    return Constraints.empty().withEmail(new Email());
  }

  public static Constraints ofDecimalMin(DecimalMin decimalMin) {
    return Constraints.empty().withDecimalMin(decimalMin);
  }

  public static Constraints ofDecimalMax(DecimalMax decimalMax) {
    return Constraints.empty().withDecimalMax(decimalMax);
  }

  public static Constraints ofDecimalMinAndMax(DecimalMin decimalMin, DecimalMax decimalMax) {
    return Constraints.empty().withDecimalMin(decimalMin).withDecimalMax(decimalMax);
  }

  public static Constraints ofPropertiesCount(PropertyCount propertyCount) {
    return ConstraintsBuilder.create().andOptionals().propertyCount(propertyCount).build();
  }

  public static Constraints ofMultipleOf(MultipleOf multipleOf) {
    return ConstraintsBuilder.create().andOptionals().multipleOf(multipleOf).build();
  }

  public Constraints withMin(Min min) {
    return new Constraints(
        Optional.ofNullable(min),
        max,
        decimalMin,
        decimalMax,
        size,
        pattern,
        email,
        propertyCount,
        multipleOf);
  }

  public Constraints withMax(Max max) {
    return new Constraints(
        min,
        Optional.ofNullable(max),
        decimalMin,
        decimalMax,
        size,
        pattern,
        email,
        propertyCount,
        multipleOf);
  }

  public Constraints withSize(Size size) {
    return new Constraints(
        min,
        max,
        decimalMin,
        decimalMax,
        Optional.ofNullable(size),
        pattern,
        email,
        propertyCount,
        multipleOf);
  }

  public Constraints withPattern(Pattern pattern) {
    return new Constraints(
        min,
        max,
        decimalMin,
        decimalMax,
        size,
        Optional.ofNullable(pattern),
        email,
        propertyCount,
        multipleOf);
  }

  public Constraints withEmail(Email email) {
    return new Constraints(
        min,
        max,
        decimalMin,
        decimalMax,
        size,
        pattern,
        Optional.ofNullable(email),
        propertyCount,
        multipleOf);
  }

  public Constraints withDecimalMin(DecimalMin decimalMin) {
    return new Constraints(
        min,
        max,
        Optional.ofNullable(decimalMin),
        decimalMax,
        size,
        pattern,
        email,
        propertyCount,
        multipleOf);
  }

  public Constraints withDecimalMax(DecimalMax decimalMax) {
    return new Constraints(
        min,
        max,
        decimalMin,
        Optional.ofNullable(decimalMax),
        size,
        pattern,
        email,
        propertyCount,
        multipleOf);
  }

  public Constraints and(Constraints other) {
    return new Constraints(
        Optionals.or(min, min),
        Optionals.or(max, other.max),
        Optionals.or(decimalMin, other.decimalMin),
        Optionals.or(decimalMax, other.decimalMax),
        Optionals.or(size, other.size),
        Optionals.or(pattern, other.pattern),
        Optionals.or(email, other.email),
        Optionals.or(propertyCount, other.propertyCount),
        Optionals.or(multipleOf, other.multipleOf));
  }

  public Optional<Min> getMin() {
    return min;
  }

  public Optional<Max> getMax() {
    return max;
  }

  public Optional<DecimalMin> getDecimalMin() {
    return decimalMin;
  }

  public Optional<DecimalMax> getDecimalMax() {
    return decimalMax;
  }

  public Optional<Size> getSize() {
    return size;
  }

  public Optional<Pattern> getPattern() {
    return pattern;
  }

  public Optional<Email> getEmail() {
    return email;
  }

  public <R> Optional<R> onMinFn(Function<Min, R> onMin) {
    return min.map(onMin);
  }

  public <R> Optional<R> onMaxFn(Function<Max, R> onMax) {
    return max.map(onMax);
  }

  public <R> Optional<R> onDecimalMinFn(Function<DecimalMin, R> onDecimalMin) {
    return decimalMin.map(onDecimalMin);
  }

  public <R> Optional<R> onDecimalMaxFn(Function<DecimalMax, R> onDecimalMax) {
    return decimalMax.map(onDecimalMax);
  }

  public <R> Optional<R> onSizeFn(Function<Size, R> onSize) {
    return size.map(onSize);
  }

  public <R> Optional<R> onPatternFn(Function<Pattern, R> onPattern) {
    return pattern.map(onPattern);
  }
}
