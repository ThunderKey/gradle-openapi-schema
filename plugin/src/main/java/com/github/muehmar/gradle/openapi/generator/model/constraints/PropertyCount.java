package com.github.muehmar.gradle.openapi.generator.model.constraints;

import java.util.Optional;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class PropertyCount {
  private final Optional<Integer> min;
  private final Optional<Integer> max;

  private PropertyCount(Optional<Integer> min, Optional<Integer> max) {
    this.min = min;
    this.max = max;
  }

  public static PropertyCount ofMinProperties(int min) {
    return new PropertyCount(Optional.of(min), Optional.empty());
  }

  public static PropertyCount ofMaxProperties(int max) {
    return new PropertyCount(Optional.empty(), Optional.of(max));
  }

  public static PropertyCount ofMinAndMaxProperties(int min, int max) {
    return new PropertyCount(Optional.of(min), Optional.of(max));
  }

  public Optional<Integer> getMinProperties() {
    return min;
  }

  public Optional<Integer> getMaxProperties() {
    return max;
  }
}
