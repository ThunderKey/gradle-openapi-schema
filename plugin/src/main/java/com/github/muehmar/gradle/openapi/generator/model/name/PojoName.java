package com.github.muehmar.gradle.openapi.generator.model.name;

import lombok.EqualsAndHashCode;

/** Represents the name of a pojo. */
@EqualsAndHashCode
public class PojoName {
  private final Name name;
  private final String suffix;

  PojoName(Name name, String suffix) {
    this.name = name;
    this.suffix = suffix;
  }

  public static PojoName ofName(Name name) {
    return new PojoName(name.startUpperCase(), "");
  }

  public static PojoName ofNameAndSuffix(Name name, String suffix) {
    return new PojoName(name.startUpperCase(), suffix);
  }

  public static PojoName ofNameAndSuffix(String name, String suffix) {
    return new PojoName(Name.ofString(name).startUpperCase(), suffix);
  }

  public PojoName deriveMemberSchemaName(Name pojoMemberName) {
    final Name derivedName = name.startUpperCase().append(pojoMemberName.startUpperCase());
    return new PojoName(derivedName, suffix);
  }

  public PojoName appendToName(String append) {
    return new PojoName(name.append(append), suffix);
  }

  public PojoName prependSuffix(String prependText) {
    return new PojoName(name, prependText + suffix);
  }

  public boolean equalsIgnoreCase(PojoName other) {
    return name.equalsIgnoreCase(other.name) && suffix.equals(other.suffix);
  }

  public Name getName() {
    return name;
  }

  public String getSuffix() {
    return suffix;
  }

  public String asString() {
    return name.append(suffix).asString();
  }

  @Override
  public String toString() {
    return asString();
  }
}
