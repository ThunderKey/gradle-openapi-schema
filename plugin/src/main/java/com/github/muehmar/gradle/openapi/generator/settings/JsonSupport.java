package com.github.muehmar.gradle.openapi.generator.settings;

import ch.bluecare.commons.data.PList;
import java.io.Serializable;
import java.util.Optional;

public enum JsonSupport implements Serializable {
  JACKSON("jackson"),
  NONE("none");

  private final String value;

  JsonSupport(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  public static Optional<JsonSupport> fromString(String value) {
    return PList.of(values()).find(support -> support.value.equalsIgnoreCase(value));
  }
}
