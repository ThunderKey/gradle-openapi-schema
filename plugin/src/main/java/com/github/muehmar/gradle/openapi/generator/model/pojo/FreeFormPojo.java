package com.github.muehmar.gradle.openapi.generator.model.pojo;

import com.github.muehmar.gradle.openapi.generator.model.Pojo;
import com.github.muehmar.gradle.openapi.generator.model.PojoName;
import com.github.muehmar.gradle.openapi.generator.model.Type;
import java.util.function.Function;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class FreeFormPojo implements Pojo {
  private final PojoName name;
  private final String description;

  private FreeFormPojo(PojoName name, String description) {
    this.name = name;
    this.description = description;
  }

  public static FreeFormPojo of(PojoName name, String description) {
    return new FreeFormPojo(name, description);
  }

  @Override
  public PojoName getName() {
    return name;
  }

  @Override
  public String getDescription() {
    return description;
  }

  @Override
  public Pojo addObjectTypeDescription(PojoName objectTypeName, String description) {
    return this;
  }

  @Override
  public Pojo inlineObjectReference(
      PojoName referenceName, String referenceDescription, Type referenceType) {
    return this;
  }

  @Override
  public <T> T fold(
      Function<ObjectPojo, T> onObjectPojo,
      Function<ArrayPojo, T> onArrayType,
      Function<EnumPojo, T> onEnumPojo,
      Function<ComposedPojo, T> onComposedPojo,
      Function<FreeFormPojo, T> onFreeFormPojo) {
    return onFreeFormPojo.apply(this);
  }
}
