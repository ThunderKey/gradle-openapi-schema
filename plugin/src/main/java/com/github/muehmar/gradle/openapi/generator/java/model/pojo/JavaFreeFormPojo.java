package com.github.muehmar.gradle.openapi.generator.java.model.pojo;

import com.github.muehmar.gradle.openapi.generator.java.model.JavaPojo;
import com.github.muehmar.gradle.openapi.generator.java.model.JavaPojoMember;
import com.github.muehmar.gradle.openapi.generator.java.model.type.JavaMapType;
import com.github.muehmar.gradle.openapi.generator.java.model.type.JavaType;
import com.github.muehmar.gradle.openapi.generator.model.Name;
import com.github.muehmar.gradle.openapi.generator.model.Necessity;
import com.github.muehmar.gradle.openapi.generator.model.Nullability;
import com.github.muehmar.gradle.openapi.generator.model.PojoName;
import com.github.muehmar.gradle.openapi.generator.model.Type;
import com.github.muehmar.gradle.openapi.generator.model.constraints.Constraints;
import com.github.muehmar.gradle.openapi.generator.model.pojo.FreeFormPojo;
import com.github.muehmar.gradle.openapi.generator.model.type.MapType;
import com.github.muehmar.gradle.openapi.generator.model.type.NoType;
import com.github.muehmar.gradle.openapi.generator.model.type.StringType;
import com.github.muehmar.gradle.openapi.generator.settings.TypeMappings;
import java.util.function.Function;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class JavaFreeFormPojo implements JavaPojo {
  private final PojoName name;
  private final String description;
  private final Constraints constraints;
  private static final Type VALUE_TYPE = NoType.create();

  private JavaFreeFormPojo(PojoName name, String description, Constraints constraints) {
    this.name = name;
    this.description = description;
    this.constraints = constraints;
  }

  public static JavaFreeFormPojo wrap(FreeFormPojo freeFormPojo) {
    return new JavaFreeFormPojo(
        freeFormPojo.getName(), freeFormPojo.getDescription(), freeFormPojo.getConstraints());
  }

  @Override
  public PojoName getName() {
    return name;
  }

  @Override
  public String getDescription() {
    return description;
  }

  public JavaPojoMember getMember() {
    return JavaPojoMember.of(
        Name.ofString("values"), "", getMemberType(), Necessity.REQUIRED, Nullability.NOT_NULLABLE);
  }

  public JavaType getMemberType() {
    return JavaMapType.wrap(
        MapType.ofKeyAndValueType(StringType.noFormat(), VALUE_TYPE), TypeMappings.empty());
  }

  public JavaType getValueType() {
    return JavaType.wrap(VALUE_TYPE, TypeMappings.empty());
  }

  @Override
  public <T> T fold(
      Function<JavaArrayPojo, T> onArrayPojo,
      Function<JavaEnumPojo, T> onEnumPojo,
      Function<JavaObjectPojo, T> onObjectPojo,
      Function<JavaComposedPojo, T> onComposedPojo,
      Function<JavaFreeFormPojo, T> onFreeFormPojo) {
    return onFreeFormPojo.apply(this);
  }
}
