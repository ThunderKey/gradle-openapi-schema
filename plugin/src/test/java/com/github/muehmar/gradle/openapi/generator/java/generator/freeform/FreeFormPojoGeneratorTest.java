package com.github.muehmar.gradle.openapi.generator.java.generator.freeform;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.github.muehmar.gradle.openapi.generator.java.model.pojo.JavaFreeFormPojo;
import com.github.muehmar.gradle.openapi.generator.model.PojoName;
import com.github.muehmar.gradle.openapi.generator.model.constraints.Constraints;
import com.github.muehmar.gradle.openapi.generator.model.constraints.PropertyCount;
import com.github.muehmar.gradle.openapi.generator.model.pojo.FreeFormPojo;
import com.github.muehmar.gradle.openapi.generator.settings.JsonSupport;
import com.github.muehmar.gradle.openapi.generator.settings.TestPojoSettings;
import io.github.muehmar.codegenerator.writer.Writer;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith(SnapshotExtension.class)
class FreeFormPojoGeneratorTest {

  public static final FreeFormPojo FREE_FORM_POJO =
      FreeFormPojo.of(
          PojoName.ofNameAndSuffix("FreeForm", "Dto"),
          "Free form object",
          Constraints.ofPropertiesCount(PropertyCount.ofMinAndMaxProperties(5, 10)));
  private Expect expect;

  @Test
  @SnapshotName("freeFormWithDefaultSettings")
  void generate_when_defaultSettings_then_matchSnapshot() {
    final FreeFormPojoGenerator gen = new FreeFormPojoGenerator();

    final Writer generate =
        gen.generate(
            JavaFreeFormPojo.wrap(FREE_FORM_POJO),
            TestPojoSettings.defaultSettings(),
            Writer.createDefault());

    expect.toMatchSnapshot(generate.asString());
  }

  @Test
  @SnapshotName("freeFormWithNoJsonSupport")
  void generate_when_noJsonSupport_then_matchSnapshot() {
    final FreeFormPojoGenerator gen = new FreeFormPojoGenerator();

    final Writer generate =
        gen.generate(
            JavaFreeFormPojo.wrap(FREE_FORM_POJO),
            TestPojoSettings.defaultSettings().withJsonSupport(JsonSupport.NONE),
            Writer.createDefault());

    expect.toMatchSnapshot(generate.asString());
  }

  @Test
  @SnapshotName("freeFormWithValidationDisabled")
  void generate_when_validationDisabled_then_matchSnapshot() {
    final FreeFormPojoGenerator gen = new FreeFormPojoGenerator();

    final Writer generate =
        gen.generate(
            JavaFreeFormPojo.wrap(FREE_FORM_POJO),
            TestPojoSettings.defaultSettings().withEnableValidation(false),
            Writer.createDefault());

    expect.toMatchSnapshot(generate.asString());
  }
}
