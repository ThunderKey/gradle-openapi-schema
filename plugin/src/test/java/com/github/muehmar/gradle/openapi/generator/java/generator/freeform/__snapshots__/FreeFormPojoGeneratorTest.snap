freeFormWithDefaultSettings=[
package com.github.muehmar;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.Map;
import java.util.Objects;
import java.util.Optional;
import javax.validation.constraints.NotNull;

/**
 * Free form object
 */
public final class FreeFormDto {
  @JsonValue
  private final Map<String, Object> values;

  @JsonCreator
  public FreeFormDto(
      Map<String, Object> values
    ) {
    this.values = values;
  }

  /**
   * 
   */
  @NotNull
  public Map<String, Object> getValues() {
    return values;
  }

  public Optional<Object> getProperty(String propertyName) {
    return Optional.ofNullable(values.get(propertyName));
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) return true;
    if (obj == null || this.getClass() != obj.getClass()) return false;
    final FreeFormDto other = (FreeFormDto) obj;
    return Objects.equals(values, other.values);
  }

  @Override
  public int hashCode() {
    return Objects.hash(
      values
    );
  }
}
]


freeFormWithNoJsonSupport=[
package com.github.muehmar;

import java.util.Map;
import java.util.Objects;
import java.util.Optional;
import javax.validation.constraints.NotNull;

/**
 * Free form object
 */
public final class FreeFormDto {
  private final Map<String, Object> values;

  public FreeFormDto(
      Map<String, Object> values
    ) {
    this.values = values;
  }

  /**
   * 
   */
  @NotNull
  public Map<String, Object> getValues() {
    return values;
  }

  public Optional<Object> getProperty(String propertyName) {
    return Optional.ofNullable(values.get(propertyName));
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) return true;
    if (obj == null || this.getClass() != obj.getClass()) return false;
    final FreeFormDto other = (FreeFormDto) obj;
    return Objects.equals(values, other.values);
  }

  @Override
  public int hashCode() {
    return Objects.hash(
      values
    );
  }
}
]


freeFormWithValidationDisabled=[
package com.github.muehmar;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.Map;
import java.util.Objects;
import java.util.Optional;

/**
 * Free form object
 */
public final class FreeFormDto {
  @JsonValue
  private final Map<String, Object> values;

  @JsonCreator
  public FreeFormDto(
      Map<String, Object> values
    ) {
    this.values = values;
  }

  /**
   * 
   */
  public Map<String, Object> getValues() {
    return values;
  }

  public Optional<Object> getProperty(String propertyName) {
    return Optional.ofNullable(values.get(propertyName));
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) return true;
    if (obj == null || this.getClass() != obj.getClass()) return false;
    final FreeFormDto other = (FreeFormDto) obj;
    return Objects.equals(values, other.values);
  }

  @Override
  public int hashCode() {
    return Objects.hash(
      values
    );
  }
}
]